api/src/main/java/com/datastrato/gravitino/authorization/AuthorizationRole.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+/** Authorization Role interface. */
+public interface AuthorizationRole {
```

Comment:
This interface is for developers, not for users, so I would suggest to move it to core, not in API module. Also is it better rename to `AuthorizationPlugin`, what do you think?

Reply:
DONE